src/Database/Postgres.js

```diff
@@ -4,40 +4,21 @@
 var pg = require('pg');
 
 exports["connect'"] = function (conString) {
-  return function(success, error) {
+  return function(error, success) {
     var client = new pg.Client(conString);
     client.connect(function(err) {
       if (err) {
         error(err);
       } else {
         success(client);
       }
```

Comment:
Note, that these functions still need to return something for a canceler, even if it just invokes a callback (please consult the example). If it's possible to cancel the effect, then it should be done so we don't accidentally leak resources. Does `pg.Client` support some way to close the pending connection?

Reply:
For reference: https://github.com/slamdata/purescript-aff/blob/master/src/Control/Monad/Aff/Compat.purs#L35-L38

The canceler isn't optional, even through the FFI, and it's an important part of Aff's semantics.

Reply:
Ok pushed one for `connect'` that I think is correct, but not sure about the `query` ones. I don't think those are cancellable .... does that mean to just call `onCancelerSuccess`?

Reply:
If there is no way to cancel a pending request, then yes, the only thing that can be done is call `onCancelerSuccess`.

Reply:
cool, I think this is all set then

Reply:
There is, or at least these was a difference between `withClient` and `withConnection`. `withClient` would pull a connection from the connection pool and `withConnection` would make a new connection.

Now I'm looking at the pg documentation and they have a new pooling API so I'm not sure how `client.connect` works currently.

When we've taken a connection from the pool, and we cancel the `Aff` we should try to cancel the query and return the connection to the pool (there's a pending PR to add query canceling to pg https://github.com/brianc/node-postgres/pull/1392). With single connections to the db without a pool we can just close the connection.

I can take a closer look at the pg API during the weekend, as I'm not sure the API we have here is relevant anymore.

Reply:
@anttih What do you think about passing that off to the user? Like export the `Client` and `Pool` constructors and let the user of this lib choose which to use and pass them into the query functions?

Reply:
@tonyd256 I'd like to keep it simple for now and concentrate on `Pool` only. That means we would expose the `Pool` constructor with something like `mkPool :: PoolConfig` where the `PoolConfig` would have all the config flags available. I think we can ditch the `connectionstring` for now.

We could then replace `withClient` with `connect`:

```purescript
connect :: forall eff a
   . Pool
  -> (Client -> Aff (db :: DB | eff) a)
  -> Aff (db :: DB | eff) a
```

What do you think? And thanks for working on this!

Reply:
@anttih I think focusing on Pool makes sense. I'd like to not ditch connection string though since that's the main way I'd like to interact with the api.